src/EnergyPlus/HVACUnitaryBypassVAV.cc

```diff
@@ -3502,6 +3511,7 @@ namespace HVACUnitaryBypassVAV {
         if (CBVAV(CBVAVNum).LastMode != CBVAV(CBVAVNum).HeatCoolMode) {
             CBVAV(CBVAVNum).changeOverTimer = thisTime + CBVAV(CBVAVNum).minModeChangeTime;
             CBVAV(CBVAVNum).LastMode = CBVAV(CBVAVNum).HeatCoolMode;
+            CBVAV(CBVAVNum).modeChanged = true;
```

Comment:
At the bottom of this function the timer is set and the modeChanged flag also needs to be set.